taichi/aot/module_builder.h

```diff
@@ -81,6 +88,7 @@ class AotModuleBuilder {
 
  private:
   std::unordered_map<std::string, aot::CompiledGraph> graphs_;
+  Program *prog_ = nullptr;
```

Comment:
Instead of saving a `Program*` in the base module, I wonder is saving a `LlvmProgramImpl*` inside the backend module builder enough for this use case? (And I assume we want to get rid of it in the future right? 
Saving a pointer to program is like opening a can of worms so I feel like we should limit its scope as much as possible.

Reply:
That's really good point, appreciate it!